core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,121 @@
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint  commontypes.MonitoringEndpoint
+	headBroadcaster     httypes.HeadBroadcaster
+	headCh              chan blockKey
+	customTelemChanSize uint8
+	unsubscribe         func()
+	chDone              chan struct{}
+	lggr                logger.Logger
+	version             string
+}
+
+// NewAutomationCustomTelemetryService creates a telemetry service for new blocks and node version
+func NewAutomationCustomTelemetryService(me commontypes.MonitoringEndpoint, hb httypes.HeadBroadcaster, lggr logger.Logger, vers string) *AutomationCustomTelemetryService {
+	return &AutomationCustomTelemetryService{
+		monitoringEndpoint:  me,
+		headBroadcaster:     hb,
+		headCh:              make(chan blockKey, 50),
+		customTelemChanSize: 50,
+		chDone:              make(chan struct{}),
+		lggr:                lggr,
+		version:             vers,
+	}
+}
+
+// Start starts
+func (e *AutomationCustomTelemetryService) Start(context.Context) error {
+	return e.StartOnce("AutomationCustomTelemetryService", func() error {
+		versionMsg := &telem.NodeVersion{
+			Timestamp:   uint64(time.Now().UTC().UnixMilli()),
+			NodeVersion: e.version,
+		}
+		wrappedMessage := &telem.AutomationTelemWrapper{
+			Msg: &telem.AutomationTelemWrapper_NodeVersion{
+				NodeVersion: versionMsg,
+			},
+		}
+		bytes, err := proto.Marshal(wrappedMessage)
+		if err != nil {
+			e.lggr.Errorf("Error occured while marshalling the message: %v", err)
+		}
+		e.monitoringEndpoint.SendLog(bytes)
+		e.lggr.Infof("BlockNumber Message Sent to Endpoint: %s", wrappedMessage.String())
+		_, e.unsubscribe = e.headBroadcaster.Subscribe(&headWrapper{e.headCh})
+		go func() {
+			e.lggr.Infof("Started enhanced telemetry service")
+			for {
+				select {
+				case blockKey := <-e.headCh:
+					blockNumMsg := &telem.BlockNumber{
+						Timestamp:   uint64(time.Now().UTC().UnixMilli()),
+						BlockNumber: uint64(blockKey.block),
+						BlockHash:   blockKey.hash,
+					}
+					wrappedMessage := &telem.AutomationTelemWrapper{
+						Msg: &telem.AutomationTelemWrapper_BlockNumber{
+							BlockNumber: blockNumMsg,
+						},
+					}
+					bytes, err := proto.Marshal(wrappedMessage)
+					if err != nil {
+						e.lggr.Errorf("Error occured while marshalling the message: %v", err)
+					}
+					e.monitoringEndpoint.SendLog(bytes)
+					e.lggr.Infof("BlockNumber Message Sent to Endpoint: %s", wrappedMessage.String())
+				case <-e.chDone:
+					return
+				}
+			}
+		}()
+		return nil
+	})
+}
+
+// Close stops go routines and closes channels
+func (e *AutomationCustomTelemetryService) Close() error {
+	return e.StopOnce("AutomationCustomTelemetryService", func() error {
+		e.chDone <- struct{}{}
+		close(e.headCh)
+		close(e.chDone)
+		e.unsubscribe()
+		e.lggr.Infof("Stopping custom telemetry service for job")
```

Comment:
move this to after line 92. if sth is wrong before this, we don't know if we attempted stopping service